go/ql/lib/semmle/go/dataflow/ExternalFlow.qll

```diff
@@ -87,6 +90,93 @@ private import internal.FlowSummaryImpl::Private
 private import internal.FlowSummaryImpl::Private::External
 private import codeql.mad.ModelValidation as SharedModelVal
 
+/** Gets the prefix for a group of packages. */
+string groupPrefix() { result = "group:" }
+
+/** Gets a group that `package` is in, according to `packageGrouping`. */
+private string getGroup(string package) {
```

Comment:
Suggest call this `getPackage(string packageOrGroup)`, and push the "if is group then resolve, else result = input" logic in here (with a `bindingset`, so it gets inlined at use sites).